src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,143 @@
+import React, {
+  useState,
+  useContext,
+  useRef,
+  useEffect,
+} from 'react';
+import { useClickAway } from 'react-use';
+import { TodosContext } from '../../context/TodosContext';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../enums/Status';
+
+type Props = {
+  item: Todo,
+};
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+export const TodoItem: React.FC<Props> = ({ item }) => {
+  const { todos, updateTodos } = useContext(TodosContext);
+  const [value, setValue] = useState(item.title);
+  const [completed, setCompleted] = useState(item.completed);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const hasEditingTodo = todos.some(todo => todo.completed === 'editing');
```

Comment:
The `completed` property of a todo is used to check if a todo is being edited. This is confusing because `completed` should indicate if a todo is completed or not. Consider using a different property to show if a todo item is being edited.